controller/controller.go

```diff
@@ -1138,6 +1155,33 @@ func (ctrl *ProvisionController) deleteVolumeOperation(volume *v1.PersistentVolu
 		return err
 	}
 
+	if len(newVolume.ObjectMeta.Finalizers) > 0 {
+		// Remove external-provisioner finalizer
+
+		// need to get the pv again because the delete has updated the object with a deletion timestamp
+		newVolume, err := ctrl.client.CoreV1().PersistentVolumes().Get(volume.Name, metav1.GetOptions{})
+		if err != nil {
+			// If we get an error the volume is probably gone
+			return nil
```

Comment:
we can check for apierrs.IsNotFound explicitly here as well as we do for Update below